### PR TITLE
java and jace PixelsPerformance: Write strips to match the C++ behaviour

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Match the C++ strip size, using the logic from ImageConverter.

### CODE REVIEW

**Comment by rleigh-codelibre** [comment]
I'll add the same for jace in a moment.

**Comment by dgault** [comment]
This does appear to match the CPP behaviour and builds and runs successfully.

As a minor nitpicky thing, the hardcoded 65535 could be replaced by Character.MAX_VALUE

File: src/main/java/ome/files/performance/PixelsPerformance.java [hunk 1]
```diff
@@ -156,8 +158,15 @@ public static void main(String[] args) {
...
+                    int sizeX = meta.getPixelsSizeX(series).getValue().intValue();
+                    IFD ifd = new IFD();
+                    int rows = 65535 / sizeX;
+                    if (rows < 1) {
+                      rows = 1;
+                    }
+                    ifd.put(IFD.ROWS_PER_STRIP, rows);
                     byte[] data = planes.get(plane);
-                    writer.saveBytes(plane, data);
+                    ((TiffWriter) writer).saveBytes(plane, data, ifd);
```
Comment (sbesson) [inline src/main/java/ome/files/performance/PixelsPerformance.java:170]: From https://ci.openmicroscopy.org/job/OME-FILES-performance-win/123/

```
pass 0: write series 2199: ...done
pass 0: write series 2200: .Error: caught exception: loci.formats.FormatException: File is too large; call setBigTiff(true)
loci.formats.FormatException: File is too large; call setBigTiff(true)
	at loci.formats.out.TiffWriter.prepareToWriteImage(TiffWriter.java:390)
	at loci.formats.out.TiffWriter.saveBytes(TiffWriter.java:263)
	at loci.formats.out.OMETiffWriter.saveBytes(OMETiffWriter.java:195)
	at loci.formats.out.TiffWriter.saveBytes(TiffWriter.java:204)
	at ome.files.performance.PixelsPerformance.main(PixelsPerformance.java:169)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:297)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by sbesson** [comment]
Interesting with #25 and this PR included, on beluga, we now have differing sizes for C++ vs Jace/Java

```
[sbesson@ome-c6220-1-4 out]$ ls -alh /scratch/tmp/data/out/
total 13G
drwxr-xr-x  2 root    root 4.0K Feb 13 19:58 .
drwxr-xr-x 16 sbesson lsd  4.0K Feb 13 09:13 ..
-rw-r--r--  1 root    root 3.4G Feb 13 19:55 bbbc-cpp.ome.tiff
-rw-r--r--  1 root    root 2.4M Feb 13 19:42 bbbc-cpp.ome.xml
-rw-r--r--  1 root    root 4.2G Feb 13 19:24 bbbc-jace.ome.tiff
-rw-r--r--  1 root    root 2.4M Feb 13 17:40 bbbc-jace.ome.xml
-rw-r--r--  1 root    root 4.2G Feb 13 17:35 bbbc-java.ome.tiff
-rw-r--r--  1 root    root 2.4M Feb 13 16:29 bbbc-java.ome.xml
-rw-r--r--  1 root    root 129M Feb 13 19:57 mitocheck-cpp.ome.tiff
-rw-r--r--  1 root    root 2.8M Feb 13 19:43 mitocheck-cpp.ome.xml
-rw-r--r--  1 root    root 130M Feb 13 19:26 mitocheck-jace.ome.tiff
-rw-r--r--  1 root    root 3.2M Feb 13 17:42 mitocheck-jace.ome.xml
-rw-r--r--  1 root    root 130M Feb 13 17:37 mitocheck-java.ome.tiff
-rw-r--r--  1 root    root 3.2M Feb 13 16:30 mitocheck-java.ome.xml
-rw-r--r--  1 root    root 216M Feb 13 19:58 tubhiswt-cpp.ome.tiff
-rw-r--r--  1 root    root 177K Feb 13 19:43 tubhiswt-cpp.ome.xml
-rw-r--r--  1 root    root 268M Feb 13 19:41 tubhiswt-jace.ome.tiff
-rw-r--r--  1 root    root 177K Feb 13 17:42 tubhiswt-jace.ome.xml
-rw-r--r--  1 root    root 268M Feb 13 17:40 tubhiswt-java.ome.tiff
-rw-r--r--  1 root    root 177K Feb 13 16:30 tubhiswt-java.ome.xml
```

It might be good to take a look at the difference between the layout of the various TIFFs to make sure we are measuring the same writing behavior.

**Comment by sbesson** [comment]
Otherwise, in terms of actual time, it looks like this provides a net improvement over previous values as reported in https://github.com/openmicroscopy/ome-files-performance/pull/24, e.g. for writing BBBC pixeldata, the Java times read like ~300000s vs 4500000s (Windows) and 90000s vs 140000s (Linux)

**Comment by rleigh-codelibre** [comment]
@dgault Thanks for mentioning the strip size, turns out I'd got it wrong.  Fixed up to use 65536 to match the C++ code.

**Comment by sbesson** [comment]
Everything completed successfully in https://ci.openmicroscopy.org/job/OME-FILES-performance-win/130/ and the generated OME-TIFFs now have the same size independently of the language. Thanks @dgault for the pointer and @rleigh-codelibre for the investigation.

**Pull request merged**